client/installation-helper.cc

```diff
@@ -0,0 +1,226 @@
+#include <snapper/Hooks.h>
+
+#include "utils/GetOpts.h"
+
+
+using namespace snapper;
+using namespace std;
+
+
+void
+step1(const string& device, const string& description)
+{
+    // step runs in inst-sys
+
+    cout << "step 1 device:" << device << endl;
+
+    cout << "temporarily mounting device" << endl;
+
+    SDir s_dir("/");
+
+    TmpMount tmp_mount(s_dir, device, "tmp-mnt-XXXXXX", "btrfs", 0, "");
+
+    cout << "copying config-file" << endl;
+
+    mkdir((tmp_mount.getFullname() + "/etc").c_str(), 0777);
+    mkdir((tmp_mount.getFullname() + "/etc/snapper").c_str(), 0777);
+    mkdir((tmp_mount.getFullname() + "/etc/snapper/configs").c_str(), 0777);
+
+    try
+    {
+	SysconfigFile config(CONFIGTEMPLATEDIR "/" "default");
+
+	config.setName(tmp_mount.getFullname() + CONFIGSDIR "/" "root");
+
+	config.setValue(KEY_SUBVOLUME, "/");
+	config.setValue(KEY_FSTYPE, "btrfs");
+    }
+    catch (const FileNotFoundException& e)
+    {
+	cerr << "copying config-file failed" << endl;
+    }
+
+    cout << "creating filesystem config" << endl;
+
+    Btrfs btrfs("/", tmp_mount.getFullname());
+
+    btrfs.createConfig();
+
+    cout << "creating snapshot" << endl;
+
+    Snapper snapper("root", tmp_mount.getFullname());
+
+    SCD scd;
+    scd.read_only = false;
+    scd.description = description;
+
+    Snapshots::iterator snapshot = snapper.createSingleSnapshot(scd);
+
+    cout << "setting default subvolume" << endl;
+
+    snapper.getFilesystem()->setDefault(snapshot->getNum());
+
+    cout << "done" << endl;
+}
+
+
+void
+step2(const string& device, const string& root_prefix, const string& default_subvolume_name)
+{
+    // step runs in inst-sys
+
+    cout << "step 2 device:" << device << " root-prefix:" << root_prefix
+	 << " default-subvolume-name:" << default_subvolume_name << endl;
+
+    cout << "mounting device" << endl;
+
+    string subvol_option = default_subvolume_name;
+    if (!subvol_option.empty())
+	subvol_option += "/";
+    subvol_option += ".snapshots";
+
+    SDir s_dir(root_prefix + "/.snapshots");
+    if (!s_dir.mount(device, "btrfs", 0, "subvol=" + subvol_option))
+    {
+	cerr << "mounting .snapshots failed" << endl;
+    }
+
+    cout << "done" << endl;
+}
+
+
+void
+step3(const string& root_prefix, const string& default_subvolume_name)
+{
+    // step runs in inst-sys
+
+    cout << "step 3 root-prefix:" << root_prefix << " default_subvolume_name:"
+	 << default_subvolume_name << endl;
+
+    cout << "adding .snapshots to fstab" << endl;
+
+    Btrfs btrfs("/", root_prefix);
+
+    btrfs.addToFstab(default_subvolume_name);
+
+    cout << "done" << endl;
+}
+
+
+void
+step4()
+{
+    // step runs in chroot
+
+    cout << "step 4" << endl;
+
+    cout << "modifying sysconfig-file" << endl;
+
+    try
+    {
+	SysconfigFile sysconfig(SYSCONFIGFILE);
+	sysconfig.setValue("SNAPPER_CONFIGS", { "root" });
+    }
+    catch (const FileNotFoundException& e)
+    {
+	cerr << "sysconfig-file not found" << endl;
```

Comment:
so if I get code it is not fatal to not modify sysconfig and it can continue with other settings, right?